test/jdk/sun/security/ssl/X509TrustManagerImpl/distrust/Camerfirma.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import java.security.Security;
+import java.time.*;
+import java.util.*;
+import javax.net.ssl.*;
+
+/**
+ * @test
+ * @bug 8346587
+ * @summary Check that TLS Server certificates chaining back to distrusted
+ *          Camerfirma roots are invalid
+ * @library /test/lib
+ * @modules java.base/sun.security.validator
+ * @run main/othervm Camerfirma after policyOn invalid
+ * @run main/othervm Camerfirma after policyOff valid
+ * @run main/othervm Camerfirma before policyOn valid
+ * @run main/othervm Camerfirma before policyOff valid
+ */
+
+public class Camerfirma {
+
+    private static final String certPath = "chains" + File.separator + "camerfirma";
+
+    // Each of the roots have a test certificate chain stored in a file
+    // named "<root>-chain.pem".
+    private static String[] rootsToTest = new String[] {
+            "camerfirmachamberscommerceca", "camerfirmachambersca",
+            "camerfirmachambersignca"};
+
+    // Date after the restrictions take effect
+    private static final ZonedDateTime DISTRUST_DATE =
+            LocalDate.of(2025, 04, 16).atStartOfDay(ZoneOffset.UTC);
+
+    public static void main(String[] args) throws Exception {
+        String prop = Security.getProperty("jdk.certpath.disabledAlgorithms");
```

Comment:
Can you add a comment here saying that some (all?) of the test certificates are signed with SHA-1 so we need to remove the constraint that disallows SHA-1 certificates?

Reply:
fixed